core/CMakeLists.txt

```diff
@@ -124,10 +124,48 @@ target_include_directories(tangram-core
   deps/double-conversion/include
 )
 
+# Add JavaScript implementation.
+if(TANGRAM_JSCORE_ENABLED)
+
+  target_compile_definitions(tangram-core
+    PRIVATE
+    TANGRAM_USE_JSCORE=1)
+
+  target_sources(tangram-core PRIVATE src/js/JSCoreContext.cpp)
+
+  if (TANGRAM_USE_JSCORE_STATIC)
+    if(NOT EXISTS "${PROJECT_SOURCE_DIR}/deps/jscore/lib/libJavaScriptCore.a")
+      message(SEND_ERROR "Please add static JavaScriptCore libary to core/deps/jscore/")
+    endif()
+    find_package(ICU COMPONENTS io i18n REQUIRED)
+    link_directories(${ICU_LIBRARY_DIRS})
+    target_link_libraries(tangram-core
+      PRIVATE
+      ${ICU_LIBRARIES}
+      "${PROJECT_SOURCE_DIR}/deps/jscore/lib/libJavaScriptCore.a"
+      "${PROJECT_SOURCE_DIR}/deps/jscore/lib/libWTF.a"
+      "${PROJECT_SOURCE_DIR}/deps/jscore/lib/libbmalloc.a")
+    target_include_directories(tangram-core
+      PRIVATE
+      "${PROJECT_SOURCE_DIR}/deps/jscore/include")
```

Comment:
@hjanetzek I take it this is for you to test JSCore on linux - Is there any way this can be simplified or moved so that it clutters the file a bit less? Can we `find_package` on an installed JavaScriptCore package instead?

Reply:
Something like https://packages.ubuntu.com/bionic/libjavascriptcoregtk-4.0-dev ? I would try it myself but I'm not on my Ubuntu machine.

Reply:
But yeah also it's probably not going to be frequently used and requires providing other files anyway, so maybe the build configuration could be kept externally too.